bodhi/models/models.py

```diff
@@ -1077,6 +1077,16 @@ def set_request(self, db, action, username):
                 update=self, agent=username))
             return
 
+        # If an update with pending status receives negative karma, make sure it gets obsoleted
+        if self.status is UpdateStatus.pending and self.request is UpdateRequest.testing \
+            and self.karma < 0:
```

Comment:
@trishnaguha I think just checking that method should be fine. Does that method not account for users who changed their minds? If it doesn't, let's file a separate issue (and separate PR) for that and we can fix it later.